src/toniarts/openkeeper/game/controller/creature/CreatureController.java

```diff
@@ -858,7 +858,21 @@ public void resetReEvaluationTimer() {
     @Override
     public void addGold(int amount) {
         Gold gold = entityData.getComponent(entityId, Gold.class);
-        entityData.setComponent(entityId, new Gold(gold.gold + amount, gold.maxGold));
+
+        // Drop excess gold, we can only carry so much
+        boolean hasMaxGold = gold.maxGold > 0;
+        int maxGoldCanAdd = hasMaxGold ? gold.maxGold - gold.gold : Integer.MAX_VALUE;

```

Comment:
Why isn't MAX_VALUE used to designate no limit in the first place?

Reply:
That is a question I ask myself every night I go to sleep... :D Yeah, I'll change it. The logic is still a bit iffy here anyway, I didn't even check that do normal creatures have a max value they can carry? Essentially dropping all bribes and pay they are given if they didn't have a chance to blow it all in the rigged casino...

Reply:
Ok, I checked. Creatures always have some max gold set, everybody. 0 is not used to indicate max gold. That was my half assed assumption. So removed all assumptions from here.

Also normal creatures do have max gold. I guess it makes sense in the real world. But I don't know how the normal creatures deal with their possessions...